snowflake/assets/configuration/spec.yaml

```diff
@@ -0,0 +1,176 @@
+name: Snowflake
+files:
+- name: snowflake.yaml
+  options:
+  - template: init_config
+    options:
+      - name: global_custom_queries
+        description: |
+          See `custom_queries` defined below.
+          Global custom queries can be applied to all instances using the
+          `use_global_custom_queries` setting at the instance level.
+        value:
+          type: array
+          items:
+            type: object
+          default: null
+          example:
+            - metric_prefix: snowflake
+              query: <QUERY>
+              columns: <COLUMNS>
+              tags: <TAGS>
+      - template: init_config/default
+  - template: instances
+    options:
+      - name: account
+        required: true
+        description: |
+          Name of your account (provided by Snowflake), including the platform and region if applicable.
+        value:
+          type: string
+          example: <ACCOUNT>
+      - name: user
+        required: true
+        description: Login name for the user.
+        value:
+          type: string
+          example: <USER>
+      - name: password
+        required: true
+        description: Password for the user
+        value:
+          type: string
+          example: <PASSWORD>
+      - name: database
+        description: Name of the default database to use.
+        value:
+          type: string
+          example: SNOWFLAKE
+      - name: schema
+        description: Name of the default schema to use for the database.
+        value:
+          type: string
+          example: ACCOUNT_USAGE
+      - name: role
+        description: Name of the default role to use.
+        value:
+          type: string
+          example: ACCOUNTADMIN
+      - name: warehouse
+        description: Name of the default warehouse to use.
+        value:
+          type: string
+          example: <WAREHOUSE>
+      - name: passcode_in_password
+        description: |
+          Enable if the Multi-Factor Authentication passcode is embedded in the login password.
+        value:
+          type: boolean
+          example: false
+      - name: passcode
+        description: The passcode provided by Duo when using MFA for login.
```

Comment:
The MFA passcode will invalidate at some point. Doesn't that create issues?

Reply:
Revisiting the different auth methods, I'm removing this option as it doesn't seem recommended either.